packages/shared/lib/services/notification/error.service.ts

```diff
@@ -0,0 +1,91 @@
+import type { Knex } from 'knex';
+import type { ActiveLog } from '@nangohq/types';
+import { Ok, Err } from '@nangohq/utils';
+import type { Result } from '@nangohq/utils';
+
+import db from '../../db/database.js';
+
+const DB_TABLE = '_nango_active_logs';
+
+type ErrorNotification = Required<Pick<ActiveLog, 'type' | 'action' | 'connection_id' | 'activity_log_id' | 'log_id' | 'active'>>;
+type SyncErrorNotification = ErrorNotification & Required<Pick<ActiveLog, 'sync_id'>>;
+
+export const errorNotificationService = {
+    auth: {
+        create: async ({ type, action, connection_id, activity_log_id, log_id, active }: ErrorNotification): Promise<Result<ActiveLog>> => {
+            return await db.knex.transaction(async (trx) => {
+                await errorNotificationService.auth.clear({ connection_id, trx });
+                const created = await trx
+                    .from<ActiveLog>(DB_TABLE)
+                    .insert({
+                        type,
+                        action,
+                        connection_id,
+                        activity_log_id,
+                        log_id,
+                        active
+                    })
+                    .returning('*');
+
+                if (created?.[0]) {
+                    return Ok(created[0]);
+                } else {
+                    return Err('Failed to create notification');
+                }
+            });
+        },
+        get: async (id: number): Promise<ActiveLog | null> => {
+            return await db.knex.from<ActiveLog>(DB_TABLE).where({ type: 'auth', connection_id: id, active: true }).first();
+        },
+        clear: async ({ connection_id, trx }: { connection_id: ActiveLog['connection_id']; trx?: Knex.Transaction }): Promise<void> => {
+            if (trx) {
+                await trx.from<ActiveLog>(DB_TABLE).where({ type: 'auth', connection_id }).delete();
+            } else {
+                await db.knex.from<ActiveLog>(DB_TABLE).where({ type: 'auth', connection_id }).delete();
+            }
+        }
```

Comment:
note that you can swap them transparently, with just a little bit of type help 
```suggestion
        clear: async ({ connection_id, trx = db.knex }: { connection_id: ActiveLog['connection_id']; trx?: Knex.Transaction | Knex }): Promise<void> => {
            await trx.from<ActiveLog>(DB_TABLE).where({ type: 'auth', connection_id }).delete();
        }
```

Reply:
Thanks, updated with https://github.com/NangoHQ/nango/pull/2222/commits/e92debc5f97efd7a3dc7b4e85259f9da3bbc9052